proxy/StatPages.cc

```diff
@@ -83,7 +83,7 @@ StatPagesManager::handle_http(Continuation *cont, HTTPHdr *header)
     host_len       = unescapifyStr(host);
 
     for (i = 0; i < n_stat_pages; i++) {
-      if (ptr_len_cmp(host, host_len, stat_pages[i].module) == 0) {
+      if ((host_len == sizeof(stat_pages[i].module)) && (memcmp(host, stat_pages[i].module, host_len) == 0)) { 
```

Comment:
``sizeof(stat_pages[i].module)`` is giving you the size of the pointer, not the length of the string. You would have to do something like this:
```C
host_len == strlen(stat_pages[i].module) && memcmp(host, stat_pages[i].module, host_len) == 0
```